src/overlays/actors/ovl_Bg_Mori_Idomizu/z_bg_mori_idomizu.c

```diff
@@ -20,23 +35,156 @@ const ActorInit Bg_Mori_Idomizu_InitVars = {
+
+void BgMoriIdomizu_Action(BgMoriIdomizu* this, GlobalContext* globalCtx) {
+    s8 roomNum;
+    Actor* thisx = &this->actor;
+    s32 newSwitchFlag;
+
+    roomNum = globalCtx->roomCtx.curRoom.num;
+    newSwitchFlag = Flags_GetSwitch(globalCtx, thisx->params & 0x3F);
+    if (newSwitchFlag != 0) {
+        this->waterLevel = -282.0f;
+    } else {
+        this->waterLevel = 184.0f;
+    }
+    if (newSwitchFlag && !this->switchFlag) {
+        func_800800F8(globalCtx, 0xCA8, 0x46, thisx, 0);
+        this->drainTimer = 90;
+    } else if (!newSwitchFlag && this->switchFlag) {
+        func_800800F8(globalCtx, 0xCA8, 0x46, thisx, 0);
+        this->drainTimer = 90;
+        thisx->posRot.pos.y = 0.0f;
+    }
+    this->drainTimer--;
+    if ((roomNum == 7) || (roomNum == 8) || (roomNum == 9)) {
+        if (this->drainTimer < 70) {
+            Math_ApproxF(&thisx->posRot.pos.y, this->waterLevel, 3.5f);
+            BgMoriIdomizu_SetWaterLevel(globalCtx, thisx->posRot.pos.y);
+            if (this->drainTimer > 0) {
+                if (newSwitchFlag) {
+                    func_800788CC(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+                } else {
+                    func_800788CC(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG);
+                }
+            }
+        }
+    } else {
+        thisx->posRot.pos.y = this->waterLevel;
+        BgMoriIdomizu_SetWaterLevel(globalCtx, thisx->posRot.pos.y);
+        Actor_Kill(thisx);
+        return;
+    }
+    this->switchFlag = newSwitchFlag;
+}
+
+void BgMoriIdomizu_Update(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgMoriIdomizu* this = THIS;
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, globalCtx2);
+    }
+}
+
+void BgMoriIdomizu_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriIdomizu* this = THIS;
+    s32 temp2y;
+    u32 gFrames = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/BgMoriIdomizu_Init.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_mori_idomizu.c", 356);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/BgMoriIdomizu_Destroy.s")
+    func_80093D84(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A47D8.s")
+    gSPMatrix(oGfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_mori_idomizu.c", 360),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A47FC.s")
+    gSPSegment(oGfxCtx->polyXlu.p++, 0x08, globalCtx->objectCtx.status[this->objBankIndex].segment);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4850.s")
+    gDPSetEnvColor(oGfxCtx->polyXlu.p++, 0, 0, 0, 128);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Idomizu/func_808A4874.s")
+    temp2y = gFrames;
+    temp2y &= 0x7F;
```

Comment:
time for the modulus match rng
Does `%= 128` match?

Reply:
`temp2y` seems very fake either way. There is probably a way to not need a temp (by doing stuff like `* 1` maybe)

Reply:
The way not to use a temp is `%0x80` for some and `&0x7F` for others. why.